source/server-discovery-and-monitoring/server-monitoring.rst

```diff
@@ -697,7 +697,7 @@ The event API here is assumed to be like the standard `Python Event
                 topology.onServerDescriptionChanged(description, connection pool for server)
                 if description.error != Null:
                     # Clear the connection pool only after the server description is set to Unknown.
-                    clear connection pool for server
+                    clear(closeInUseConnections: isNetworkTimeout(description.error)) connection pool for server
```

Comment:
Done, with a note that I didn't validate `closeInUseConnections=true` since we have no guarantees about `PoolClear` and `ConnectionClosed` events order. I've added a different test in `pool-clear-schedule-run-smoke` to validate this flag

Reply:
This test also can contain this check if we implement a new test step like `AssertEventContent` (or similar), but not sure whether it's worth doing it since it gives some questions like what should we do if a number of PoolClear events more than one and they're different? In my opinion check on the pool tests level is enough here

Reply:
>Done, with a note that I didn't validate closeInUseConnections=true since we have no guarantees about PoolClear and ConnectionClosed events order. I've added a different test in pool-clear-schedule-run-smoke to validate this flag

The PoolClear should always come before the ConnectionClosed due to the fact that the clearing is the cause of the connection closing. Regardless, the way the SDAM test is currently written doesn't require ordering, so that shouldn't affect our ability to assert on `closeInUseConnections=true` either. I've already left some comments on that test but we should definitely be asserting on `closeInUseConnections=true`.

Also, it's not clear to me why `pool-clear-schedule-run-smoke` needs to be a different test. Let's just add a `closeInUseConnections: true` assertion to the existing `pool-clear-shecdule-run` and remove the smoke version.

Reply:
>The PoolClear should always come before the ConnectionClosed due to the fact that the clearing is the cause of the connection closing. 

it's not guaranteed. We discussed it here:  https://github.com/mongodb/specifications/pull/1170#discussion_r860061209. Yes, in most cases it will be: `PoolClear` => `ConnectionClosed`, but sometimes it's not which gives some race issues

Reply:
>Also, it's not clear to me why pool-clear-schedule-run-smoke needs to be a different test.

to be able to validate PoolCleared event without race issues with `ConnectionClosed`. That issue presents in not smoke test

Reply:
Can you elaborate on what exactly the race is? From my understanding, the pool is cleared (emits an event) and then it closes all of the in-use connections async afterwards. How would a connection get closed before the pool is cleared?

Reply:
At least in c# implementation, the pool clear event is emitted when all steps inside clear is done (otherwise if sending a prune request will fail and actual Clear method won't be fully finished, we will still have pool cleared event). So effectively happened steps are:
* Send Prune requesting with closeInUse=true (not actual connection close)
* Emit PoolClear event
in most of cases, PoolClear event is emitted before any connection become closed, but it's not guarantied and sometimes connection closing happens before PoolClear event

Reply:
I think the C# implementation should be emitting the event immediately after incrementing the generation but before taking any additional steps. From the spec:

>If a user is subscribed to Connection Monitoring events, a PoolClearedEvent MUST be emitted after incrementing the generation / marking the pool as "paused". If the pool is already "paused" when it is cleared, then the pool MUST NOT emit a PoolCleared event.

The sections that come _after_ this requirement describe the other steps the pool needs to take, clearing in use connections being one of them. Unfortunately the section lacks psuedocode to make this explicit, but I think the wording and ordering of the paragraphs is enough to convey it.

Once this change is made the race should not be an issue.

Reply:
done